src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarRegionSwitcher.test.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { BrowserNavBarRegionSwitcher } from './BrowserNavBarRegionSwitcher';
+
+import BrowserRegionEditor from '../browser-region-editor/BrowserRegionEditor';
+import BrowserRegionField from '../browser-region-field/BrowserRegionField';
+
+jest.mock(
+  'src/content/app/browser/browser-region-editor/BrowserRegionEditor',
+  () => () => <div>BrowserRegionEditor</div>
+);
+jest.mock(
+  'src/content/app/browser/browser-region-field/BrowserRegionField',
+  () => () => <div>BrowserRegionField</div>
+);
+
+const props = {
+  toggleRegionEditorActive: jest.fn(),
+  toggleRegionFieldActive: jest.fn()
+};
+
+describe('BrowserNavBarRegionSwitcher', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
```

Comment:
I don't think this is necessary here.

Reply:
The "calls cleanup functions on unmount" test calls `props.toggleRegionEditorActive` and `props.toggleRegionFieldActive`. If at any point we add another test that checks anything related to those functions, they will have already been called, and this may interfere with test results.

Example: delete the afterEach function and add another "calls cleanup functions on unmount" test after the first one. The assertions `expect(props.toggleRegionEditorActive).not.toHaveBeenCalled();` and `expect(props.toggleRegionFieldActive).not.toHaveBeenCalled();` will now fail, because these functions will have been called in the previous test.

Reply:
> If at any point we add another test that checks anything related to those functions, they will have already been called, and this may interfere with test results.

That is true but do we really need this to be there upfront? There are chances that we might never come back to update these tests.

Reply:
Between having tests that don't clean up after themselves (and do not cause problems because there are no further tests to take advantage of the clean environment), and cleaning up the definitely polluted testing environment even though there are no further tests to run, I think it's better to err on the side of the second. After all, what harm can cleaning up do — add a millisecond or two to the execution time of the test? Well-behaved tests, like good boy scouts, try to leave their environment clean.

Reply:
@azangru I agree. It is always better to have something that will avoid things breaking rather than adding it when it only breaks.